functorch/experimental/cond.py

```diff
@@ -18,6 +18,8 @@
 
 def trace_cond(proxy_mode, func_overload, pred, true_fn, false_fn, operands):
     def _unwrap_proxy(e):
+        if not isinstance(e, (torch.Tensor, torch.SymInt, torch.SymFloat)):
+            return e
```

Comment:
Is this a bug fix? How was it caught?

Reply:
The unit tests. It's not strictly a bug, but I made the asserts for proxy retrieval more strict and cond was sending us garbage previously.